src/pybind/mgr/dashboard/frontend/src/app/shared/api/auth.service.ts

```diff
@@ -42,12 +42,14 @@ export class AuthService {
   logout(callback: Function = null) {
     return this.http.post('api/auth/logout', null).subscribe((resp: any) => {
       this.authStorageService.remove();
-      const url = _.get(this.route.snapshot.queryParams, 'returnUrl', '/login');
+      if (resp.sso) {
+        return window.location.replace(resp.redirect_url);
```

Comment:
Looks like [Angular provides a specific method for this](https://angular.dev/api/common/Location). It's always better to use the Angular services, as they might be doing extra stuff (for example, I think that the `window.location.*` might override the Angular routing mechanisms).

Angular still [suggests using `Router.navigate` instead](https://angular.dev/api/common/Location?tab=usage-notes).

Reply:
What I'm trying to do here is to redirect to the backend's @Rotuer(/auth/oauth2), tried different angualar router methods, but none did the trick.

The page always ends up in a loop in the /login page, trying to log again. I think this is due to blocklisting the token on the logout process and removing the localStorage user's info so all following operations are not permitted.

So I tried to not altear any of the current implementation and just added the window.location.replace for the oauth2 logout mechanism.

Reply:
Ok!